storm-core/src/jvm/backtype/storm/blobstore/ClientBlobStore.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.blobstore;
+
+import backtype.storm.Config;
+import backtype.storm.daemon.Shutdownable;
+import backtype.storm.generated.AccessControl;
+import backtype.storm.generated.AccessControlType;
+import backtype.storm.generated.AuthorizationException;
+import backtype.storm.generated.ReadableBlobMeta;
+import backtype.storm.generated.SettableBlobMeta;
+import backtype.storm.generated.KeyAlreadyExistsException;
+import backtype.storm.generated.KeyNotFoundException;
+import backtype.storm.utils.NimbusClient;
+import backtype.storm.utils.Utils;
+
+import java.util.Iterator;
+import java.util.Map;
+
+
+public abstract class ClientBlobStore implements Shutdownable {
+  protected Map conf;
+
+  public abstract void prepare(Map conf);
+  protected abstract AtomicOutputStream createBlobToExtend(String key, SettableBlobMeta meta) throws AuthorizationException, KeyAlreadyExistsException;
+  public abstract AtomicOutputStream updateBlob(String key) throws AuthorizationException, KeyNotFoundException;
+  public abstract ReadableBlobMeta getBlobMeta(String key) throws AuthorizationException, KeyNotFoundException;
+  protected abstract void setBlobMetaToExtend(String key, SettableBlobMeta meta) throws AuthorizationException, KeyNotFoundException;
+  public abstract void deleteBlob(String key) throws AuthorizationException, KeyNotFoundException;
+  public abstract InputStreamWithMeta getBlob(String key) throws AuthorizationException, KeyNotFoundException;
+  public abstract Iterator<String> listKeys();
+  public abstract void watchBlob(String key, IBlobWatcher watcher) throws AuthorizationException;
+  public abstract void stopWatchingBlob(String key) throws AuthorizationException;
```

Comment:
All implementations of `watchBlob` and `stopWatchingBlob` either do nothing or throw RuntimeException.  Should we remove these for now?  If we have a useful implementation, we can add them back later.